### PR TITLE
[KIECLOUD-444] - enhance cekit cacher to get RHPAM/DM CRx artifacts w…

### PR DESCRIPTION
**Pull request by spolti**
…hen available

Signed-off-by: spolti <fspolti@redhat.com>

### CODE REVIEW

File: cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/builds/nightly/NightlyBuildsPullRequestAcceptor.java [hunk 1]
```diff
@@ -131,12 +110,12 @@ public void onFilePersisted(String fileName, String checkSum) {
 
                                 log.fine(String.format("Updating BC monitoring rhpam_business_central_monitoring_distribution.zip from [%s] to [%s]", artifact.getMd5(), bcMonitoringCheckSum));
                                 artifact.setMd5(bcMonitoringCheckSum);
-                                yamlFilesHelper.writeModule(bcMonitoring, bcMonitoringFile);
+                                yamlFilesHelper.writeModule( buildUtils.bcMonitoring(),  buildUtils.bcMonitoringFile());
```
Comment (jakubschwan) [inline cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/builds/nightly/NightlyBuildsPullRequestAcceptor.java:113]: formatting change: removing white spaces
```suggestion
                                yamlFilesHelper.writeModule(buildUtils.bcMonitoring(), buildUtils.bcMonitoringFile());
```

File: cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/builds/nightly/NightlyBuildsPullRequestAcceptor.java [hunk 2]
```diff
@@ -131,12 +110,12 @@ public void onFilePersisted(String fileName, String checkSum) {
...
 
                                 // find name: "rhpam_business_central_monitoring_distribution.zip"
                                 // and add comment on next line : rhpam-${version}.redhat-${buildDate}-monitoring-ee7.zip
                                 // or rhpam-${version}.PAM-redhat-${buildDate}-monitoring-ee7.zip depending on PAM version
-                                reAddComment(bcMonitoringFile, "name: \"rhpam_business_central_monitoring_distribution.zip\"",
+                                buildUtils.reAddComment( buildUtils.bcMonitoringFile(), "name: \"rhpam_business_central_monitoring_distribution.zip\"",
```
Comment (jakubschwan) [inline cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/builds/nightly/NightlyBuildsPullRequestAcceptor.java:118]: formatting change: removing white space
```suggestion
                                buildUtils.reAddComment(buildUtils.bcMonitoringFile(), "name: \"rhpam_business_central_monitoring_distribution.zip\"",
```

File: cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/builds/nightly/NightlyBuildsPullRequestAcceptor.java [hunk 3]
```diff
@@ -157,12 +136,12 @@ public void onFilePersisted(String fileName, String checkSum) {
 
                                 log.fine(String.format("Updating Business Central rhpam_business_central_distribution.zip from [%s] to [%s]", artifact.getMd5(), bcCheckSum));
                                 artifact.setMd5(bcCheckSum);
-                                yamlFilesHelper.writeModule(businessCentral, businessCentralFile);
+                                yamlFilesHelper.writeModule( buildUtils.businessCentral(),  buildUtils.businessCentralFile());
```
Comment (jakubschwan) [inline cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/builds/nightly/NightlyBuildsPullRequestAcceptor.java:139]: removing white spaces, please apply this change in all needed places
```suggestion
                                yamlFilesHelper.writeModule(buildUtils.businessCentral(), buildUtils.businessCentralFile());
```

**Review by jakubschwan — COMMENTED** [review]
Looks good to me.
One suggestion for improve, could be added artifact names and file names to some constant class? This would help to update these values in future (we wouldn't need to go through all classes). Also we are using these names in CR related class and also in Nightly related class right now.

**Comment by spolti** [comment]
@jakubschwan suggestions applied, please take another look.

File: cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/utils/BuildUtils.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+    // artifacts and zip file names used on Nightly and CR builds
+    public String RHDM_ADD_ONS_DISTRIBUTION_ZIP = "rhdm_add_ons_distribution.zip";
+    public String RHDM_ADD_ONS_ZIP = "rhdm-%s-add-ons.zip";
+
+    public String RHDM_DECISION_CENTRAL_DISTRIBUTION_ZIP = "rhdm_decision_central_distribution.zip";
+    public String RHDM_DECISION_CENTRAL_EAP7_DEPLOYABLE_ZIP = "rhdm-%s-decision-central-eap7-deployable.zip";
+
+    public String RHDM_KIE_SERVER_DISTRIBUTION_ZIP = "rhdm_kie_server_distribution.zip";
+    public String RHDM_KIE_SERVER_EE8_ZIP = "rhdm-%s-kie-server-ee8.zip";
```
Comment (jakubschwan) [inline cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/utils/BuildUtils.java:36]: For RHDM are missing nightly zip constants

Comment (spolti) [reply cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/utils/BuildUtils.java]: updated :)

**Review by jakubschwan — APPROVED** [review]

**Review by msmagnanijr — APPROVED** [review]
lgtm

**Pull request merged**